org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/encoding/WorkspaceEncodingProviderTest.java

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2017 itemis AG (http://www.itemis.com) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.tests.editor.encoding;
+
+import static org.eclipse.xtext.junit4.ui.util.IResourcesSetupUtil.*;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtext.builder.nature.XtextNature;
+import org.eclipse.xtext.junit4.ui.AbstractWorkbenchTest;
+import org.eclipse.xtext.parser.IEncodingProvider;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.testlanguages.ui.internal.TestlanguagesActivator;
+import org.eclipse.xtext.ui.editor.WorkspaceEncodingProvider;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.inject.Injector;
+
+/**
+ * @author Karsten Thoms - Initial contribution and API
+ */
+public class WorkspaceEncodingProviderTest extends AbstractWorkbenchTest {
```

Comment:
I haven't tried but I'd guess that these tests have passed before the changes below, already. No?